### PR TITLE
Update routing

### PR DESCRIPTION
**Pull request by huxley-tray**
updated routing to work as server side and set up paths for admin, and blogs
had to make a small change to style comp to make run but not perfect fix yet

TODO 

reset linted files 
add gitignore 
and try again

### CODE REVIEW

File: pages/api/posts/[postId].js [hunk 1]
```diff
@@ -9,7 +9,7 @@ async function GET(req, res) {
   } = req;
 
   const row = (await db.get()).posts.find((d) => d.id == postId);
-
+console.log('here is the row', row);
```
Comment (huxley-tray) [inline pages/api/posts/[postId].js:12]: i like seeing this but can remove

File: pages/index.tsx [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { Layout } from '../src/components/layout';
 const Home: FC = () => (
   <Layout>
     <h1>Welcome!</h1>
+    <p>its me huxley </p>
```
Comment (henryhobhouse) [inline pages/index.tsx:7]: 👀 😆

File: src/components/styled.tsx [hunk 3]
```diff
@@ -42,12 +42,15 @@ const rotate = keyframes`
...
+const animationSpinner = props => css `${rotate} 1s linear infinite;`;
+
 export const Loader: FC = (props) => (
   <ImSpinner2
     {...props}
     style={{
       verticalAlign: 'middle',
-      animation: `${rotate} 1s linear infinite`,
+        // @ts-ignore 
+      animation: animationSpinner,
```
Comment (henryhobhouse) [inline src/components/styled.tsx]: Yeah. I saw this problem. It looks like react icon use styled components below the hood. 

You can get around this (without errors) by abstracting the styled components instance:

```
const rotate = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Loader = styled(ImSpinner2)`
  vertical-align: middle;
  animation: ${css`
    ${rotate} 1s linear infinite
  `};
`;
```

File: src/index.tsx [hunk 4]
```diff
@@ -1,47 +1,27 @@
...
 
 const SafeHydrate: FC = ({ children }) => (
   <div suppressHydrationWarning>
     {typeof document === 'undefined' ? null : children}
   </div>
-)
+);
 
 const App: FC = () => (
   <SafeHydrate>
```
Comment (henryhobhouse) [inline src/index.tsx]: I don't think we need the SafeHydrate component anymore.

File: src/screens/admin/Post.js [hunk 5]
```diff
@@ -1,32 +1,28 @@
...
+import { useRouter } from 'next/router';
+import usePost from '../../hooks/usePost';
+import useSavePost from '../../hooks/useSavePost';
+import useDeletePost from '../../hooks/useDeletePost';
+import PostForm from '../../components/post-form';
+import { Loader } from '../../components/styled';
 
-//
+export default function AdminView() {
+  const { query: { adminId }, push } = useRouter();
```
Comment (henryhobhouse) [inline src/screens/admin/Post.js]: Minor point however I would still think that this should be named postId as that is what it is referencing even if in the admin section. Changing the name suggests it's not a post by extension. Super minor though.

**Review by henryhobhouse — APPROVED** [review]
Nice. 

Some minor points but overall some solid extension to the work from yesterday afternoon. If game you are welcome to merge this in and I can then merge my branch with this and can go from there.

File: pages/admin/[postId].tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  );
+};
+
+export default AdminPost;
+
+export async function getServerSideProps() {
+    return {
+      props: {}, // will be passed to the page component as props
+    };
+  }
```
Comment (henryhobhouse) [inline pages/admin/[postId].tsx:86]: Not a blocker as can do later but we can remove this now as no longer needed.

File: pages/posts/[postId].tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+export default Post;
+
+
+export async function getServerSideProps(context) {
+    return {
+      props: {postId: context.params.postId}, // will be passed to the page component as props
+    };
+  }
```
Comment (henryhobhouse) [inline pages/posts/[postId].tsx:38]: As above. This is no longer needed. However not a blocker.

File: src/hooks/useCreatePost.js [hunk 8]
```diff
@@ -1,20 +1,20 @@
...
+  });
 
   const mutate = React.useCallback(async (values) => {
-    setState({ isLoading: true })
+    setState({ isLoading: true });
     try {
-      const data = axios.post('/api/posts', values).then((res) => res.data)
-      setState({ isSuccess: true, data })
+      const data = await axios.post('/api/posts', values);
+      setState({ isSuccess: true, data });
```
Comment (henryhobhouse) [inline src/hooks/useCreatePost.js:13]: This was a legit bug in that it wasn't waiting for the return. I am tempted to add it back in to see if people notice during the workshop... ;P

**Review by henryhobhouse — APPROVED** [review]

**Pull request merged**